source/navigator/MainNavigator.tsx

```diff
@@ -171,51 +172,80 @@ const CustomNavigator = ({
 const createCustomNavigator = createNavigatorFactory(CustomNavigator);
 const MainCustomNavigator = createCustomNavigator();
 
-const MainNavigator = (): JSX.Element => {
+const MainNavigator = (): JSX.Element | null => {
+  const { visit: compatibilityVisit } = useContext(AppCompatibilityContext);
   const { userAuthState } = useContext(AuthContext);
 
-  return (
-    <MainCustomNavigator.Navigator screenOptions={{ headerShown: false }}>
-      <>
-        {userAuthState === USER_AUTH_STATE.PENDING && (
-          <MainCustomNavigator.Screen name="Start" component={SplashScreen} />
-        )}
-
-        {userAuthState === USER_AUTH_STATE.SIGNED_OUT && (
+  const useIncompatibilityWarningEffect = () =>
+    // tell the compatibility to call us with current status
+    useEffect(() =>
+      compatibilityVisit({
+        incompatible: ({ updateUrl }) =>
+          Alert.alert(
+            "Mitt Helsingborg måste uppdateras",
+            "Versionen du använder av Mitt Helsingborg är för gammal",
+            [
+              {
+                text: "Hämta uppdatering",
+                onPress: () => Linking.openURL(updateUrl),
+              },
+            ]
+          ),
+      })
+    );
```

Comment:
Did some testing. This function is executed 6 times instead of only 2 if only used `useEffect` with a proper dependency array. The main reason is that useEffect does not have a dependency array, which you should have. Otherwise the useEffect will run on every re-render.